admin/lib/validator/hdf-confirm-validator.js

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+const ValidationError = require('../validation-error')
+const hdfConfirmErrorMessages = require('../errors/hdf-confirm')
+
+/**
+ * Validates HDF confirmation data
+ * @param {Object} hdfConfirmData
+ * @returns {Object}
+ */
+module.exports.validate = function (hdfConfirmData) {
+  const validationError = new ValidationError()
+  const {
+    confirm,
+    pupilDetails,
+    uniquePins,
+    staffConfirm
+  } = hdfConfirmData
+
+  if (confirm === 'Y') {
```

Comment:
@danwilliams Can you avoid the nested if by returning the validation error if confirm !== 'Y' ?
This way the code readability is better.